pkg/cmd/healthzproxy/healthzproxy.go

```diff
@@ -0,0 +1,148 @@
+	listenKey  string
+}
+
+// NewHealthzProxyCommand creates a healthz-proxy command.
+func NewHealthzProxyCommand() *cobra.Command {
+	opts := healthzOpts{
+		listenPort:   10100, // TODO random for testing need to register with proper authorities
+		etcdEndpoint: "https://localhost:2379",
+	}
+	cmd := &cobra.Command{
+		Use:   "healthz-proxy",
+		Short: "Proxy TCP /health check to etcd server without TLS authentication",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := opts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	opts.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (h *healthzOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.Uint16Var(&h.listenPort, "listen-port", h.listenPort, "Listen on this port")
+	fs.StringVar(&h.listenCert, "listen-cert", "", "secure connections to the proxy using this TLS certificate file")
+	fs.StringVar(&h.listenKey, "listen-key", "", "secure connections to the proxy using this TLS key file")
+	fs.StringVar(&h.clientCert, "client-cert", "", "proxy client requests to etcd use this TLS certificate file")
+	fs.StringVar(&h.clientKey, "client-key", "", "proxy client requests to etcd use this TLS key file")
+	fs.StringVar(&h.clientCACert, "client-cacert", "", "proxy client requests to etcd use this TLS CA certificate file")
+	fs.StringVar(&h.etcdEndpoint, "etcd-endpoint", h.etcdEndpoint, "proxy client will dial this etcd endpoint")
+}
+
+// Validate verifies the inputs.
+func (r *healthzOpts) Validate() error {
+	if r.listenPort == 0 {
+		return fmt.Errorf("--listen-port must be between 1 and 65535")
+	}
+	if r.listenKey == "" {
+		return fmt.Errorf("missing required flag: --listen-key")
+	}
+	if r.listenCert == "" {
+		return fmt.Errorf("missing required flag: --listen-cert")
+	}
+	if r.clientKey == "" {
+		return fmt.Errorf("missing required flag: --client-key")
+	}
+	if r.clientCert == "" {
+		return fmt.Errorf("missing required flag: --client-cert")
+	}
+	if r.clientCACert == "" {
+		return fmt.Errorf("missing required flag: --client-cacert")
+	}
+	if r.etcdEndpoint == "" {
+		return fmt.Errorf("missing required flag: --etcd-endpoint")
+	}
+	return nil
+}
+
+type Health struct {
+	Health string `json:"health"`
+}
+
+// Run contains the logic of the insecure-readyz command.
+func (h *healthzOpts) Run() error {
+	caCert, err := ioutil.ReadFile(h.clientCACert)
+	if err != nil {
+		return err
+	}
+	keyPair, err := tls.LoadX509KeyPair(h.clientCert, h.clientKey)
+	if err != nil {
+		return err
+	}
+	caPool := x509.NewCertPool()
+	caPool.AppendCertsFromPEM(caCert)
+
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{keyPair},
+		RootCAs:      caPool,
+	}
+	client := &http.Client{Transport: &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/healthz", func(w http.ResponseWriter, req *http.Request) {
+		resp, err := client.Get(h.etcdEndpoint + "/health")
```

Comment:
does etcd require authentication for accessing healthz?  That makes it difficult to add to a pod

Reply:
the client Get requires auth to etcd /health but the endpoint /healthz does not. I get what you're saying though and ideally, etcd would expose this directly we have a few options that are in the works upstream to allow this.